source/slang/slang-lower-to-ir.cpp

```diff
@@ -9968,9 +9969,9 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         auto subContext = nestedContextFunc.getContext();
         auto lowered = lowerFuncDeclInContext(subContext, subBuilder, decl);
 
-        auto parentStructTypeModifier = decl->findModifier<ParentStructTypeModifier>();
-        if (parentStructTypeModifier && decl->members.getCount() == 0)
-            as<IRStructType>(parentStructTypeModifier->getArg())->defaultCtor = as<IRFunc>(lowered.val);
+        IRType** maybeParentStruct = this->context->shared->ctorBaseToParentStruct.tryGetValue(decl);
+        if (maybeParentStruct && decl->members.getCount() == 0)
```

Comment:
Don't think this is needed.

Reply:
(reference to previous comment - https://github.com/shader-slang/slang/pull/3987#discussion_r1576752102)